### ISSUE TITLE
test_spdm_responder_finish_rsp Fuzzing test ASSERT and hang

### ISSUE THREAD

**Issue by Zhiqiang520** [DMTF/libspdm#336]
Verified on 3e0e1b9
```
cd libspdm
mkdir build
cd build
cmake -DARCH=x64 -DTOOLCHAIN=AFL -DTARGET=Release -DCRYPTO=mbedtls -DGCOV=ON ..
make copy_sample_key
make
cd bin
~/libspdm/build/bin$ ./test_spdm_responder_finish_rsp ./test_spdm_responder_finish_rsp_hang
ASSERT: /home/test/libspdm/library/spdm_common_lib/libspdm_com_context_data.c(1277): finished_key_ready
^C
```
The content of binary file "test_spdm_responder_finish_rsp_hang":
![image](https://user-images.githubusercontent.com/90754803/144385182-b28dcf3f-bacd-475a-90ec-38a2a56b6f50.png
)
[test_spdm_responder_finish_rsp_hang.zip](https://github.com/DMTF/libspdm/files/7640572/test_spdm_responder_finish_rsp_hang.zip)

**Comment by qizhangz** [comment]
I will check this

**Comment by qizhangz** [comment]
@Xiaohanjlll

This fuzzing test need to  set secured_message_context->finished_key_ready to TRUE as the unit test.

Could you check the fuzzing test code?

**Comment by Xiaohanjlll** [comment]
This problem has been fixed

**Comment by Xiaohanjlll** [comment]
This problem has been fixed.

**Comment by Xiaohanjlll** [comment]
This problem has been solved on 1edab81ae400b588485f0beccae789a62d50d127.

**Comment by Xiaohanjlll** [comment]
This problem has been fixed on 1edab81ae400b588485f0beccae789a62d50d127。

**Comment by Zhiqiang520** [comment]
It was fixed, now it will not assert and hang.
Verified on the latest tree https://github.com/DMTF/libspdm/commit/8eac2c489b89a6936e9e60dc30d18315c03f0594.
